walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -12,12 +12,133 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+
+
 package com.google.sps;
 
-import java.util.Collection;
+import java.util.*;
+import static java.lang.Math.toIntExact;
+
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+
+  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {    
+    // not sure why this works!! but I have to cast the request 
+    //of attendees as an int 
+    Collection<String> attendeesCollection = request.getAttendees();
+    Collection<String> optionalAttendeesCollection = request.getOptionalAttendees();
+
+    List<TimeRange> badTimes = new ArrayList<TimeRange>();
+    List<TimeRange> goodTimes = new ArrayList<TimeRange>();
+
+    // get the duration of the event, haven't implemented yet
+    long durationLong = request.getDuration();
+    int duration = (int) durationLong;
+
+    if(duration >= TimeRange.WHOLE_DAY.duration())
+    {
+        return Arrays.asList();
+    }
+
+    Set<String> participants = new HashSet<String>();
+    if(!(request.getOptionalAttendees().isEmpty()))
+    {   
+        for(String attendee: optionalAttendeesCollection)
```

Comment:
Space after "for". Here and below. Also space after "attendee" (here and below).

Reply:
Ok!